application/tests/unit_handler.js

```diff
@@ -525,12 +565,35 @@ describe('User service', () => {
 
     //usergroups
 
+    it('Try to create wrong usergroup', () => {
+      let req = {
+        payload: wrong_usergroup,
+        auth: { //headers which will be set with JWT
+          credentials: {
+            userid: userid
+          }
+        }
+      };
+      return handler.createOrUpdateUsergroup(req, (result) => {
+        // console.log(result);
+
+        expect(result.isBoom).to.equal(true);
```

Comment:
these expectations will throw an AssertionError if not met, but the test will pass anyway, as they're inside a callback, and the errors are just ignored.

Reply:
instead we need to use the `done` callback here and a try..catch block that would catch and return the error in the callback. In this particular test case actually we get a log that while 500 is expected, 422 is returned (and 422 is actually the better code, as it's an error on the client side, not some server fault). But the test passes anyway, because it's not properly handled!!

I'm not sure if many tests unrelated to this PR are not faulty as well.

Reply:
something i don't understand here is, this particular test was added in this branch, but does not seem to be related to the functionality at hand

Reply:
Good that you realised it! I fixed the error handling in this test and I will review the other tests as well (within a new jira task).
I added this partly related test because it was missing and I thought having it would be nice. ^^